### ISSUE TITLE
ImportError:  `GLIBCXX_3.4.30' not found

### ISSUE THREAD

**Issue by rvanlaar** [rstudio/reticulate#1282]
Running 
``` 
import scipy.stats as stats
```
gives the following error:

```ibstdc++.so.6
ImportError: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /home/handboek/.local/share/r-miniconda/envs/r-reticulate/lib/python3.8/site-packages/scipy/optimize/_highs/_highs_wrapper.cpython-38-x86_64-linux-gnu.so)
```

It might be a path issue. The created miniconda enviroment by reticulate is sound.
And the included `libstdc++.so.6` in the miniconda enviroment does contain the GLIBCXX_3_4.30 string.
```
/home/handboek/.local/share/r-miniconda/envs/r-reticulate/bin/python
Python 3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 06:04:18) 
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import scipy.stats as stats
>>> 
```

## reproduction:
```
> library("reticulate")
> install_miniconda()
< snip>
> py_discover_config()
python:         /home/handboek/.local/share/r-miniconda/envs/r-reticulate/bin/python
libpython:      /home/handboek/.local/share/r-miniconda/envs/r-reticulate/lib/libpython3.8.so
pythonhome:     /home/handboek/.local/share/r-miniconda/envs/r-reticulate:/home/handboek/.local/share/r-miniconda/envs/r-reticulate
version:        3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 06:04:18)  [GCC 10.3.0]
numpy:          /home/handboek/.local/share/r-miniconda/envs/r-reticulate/lib/python3.8/site-packages/numpy
numpy_version:  1.23.3
> pad <- py_discover_config()
> use_python(pad$python)
> py_install("scipy")
+ '/home/handboek/.local/share/r-miniconda/bin/conda' 'install' '--yes' '--prefix' '/home/handboek/.local/share/r-miniconda/envs/r-reticulate' '-c' 'conda-forge' 'scipy'
Collecting package metadata (current_repodata.json): done
Solving environment: done
<snip>
Downloading and Extracting Packages
scipy-1.9.1          | 26.2 MB   | ################################################## | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
Retrieving notices: ...working... done
> repl_python()
Python 3.8.13 (/home/handboek/.local/share/r-miniconda/envs/r-reticulate/bin/python)
Reticulate 1.26 REPL -- A Python interpreter in R.
Enter 'exit' or 'quit' to exit the REPL and return to R.
>>> import scipy.stats as stats
ImportError: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /home/handboek/.local/share/r-miniconda/envs/r-reticulate/lib/python3.8/site-packages/scipy/optimize/_highs/_highs_wrapper.cpython-38-x86_64-linux-gnu.so)
>>> exit
```

## environment
```
> utils::sessionInfo() 
R version 4.2.1 (2022-06-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Debian GNU/Linux 11 (bullseye)

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] reticulate_1.26

loaded via a namespace (and not attached):
 [1] compiler_4.2.1  here_1.0.1      Matrix_1.5-1    rprojroot_2.0.3
 [5] rappdirs_0.3.3  Rcpp_1.0.9      grid_4.2.1      jsonlite_1.8.0 
 [9] png_0.1-7       lattice_0.20-45
```

```
> reticulate::py_config() 
python:         /home/handboek/.local/share/r-miniconda/envs/r-reticulate/bin/python
libpython:      /home/handboek/.local/share/r-miniconda/envs/r-reticulate/lib/libpython3.8.so
pythonhome:     /home/handboek/.local/share/r-miniconda/envs/r-reticulate:/home/handboek/.local/share/r-miniconda/envs/r-reticulate
version:        3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 06:04:18)  [GCC 10.3.0]
numpy:          /home/handboek/.local/share/r-miniconda/envs/r-reticulate/lib/python3.8/site-packages/numpy
numpy_version:  1.23.3
sys:            [builtin module]

NOTE: Python version was forced by use_python function
```

**Comment by carmensandoval** [comment]
I think this might be an issue with scipy itself...
I'm getting a similar error:

```
library(reticulate)
reticulate::use_condaenv("pegasus", required = TRUE)
reticulate::py_config()

python:         /gstore/home/sandovc9/mambaforge/envs/pegasus/bin/python
libpython:      /gstore/home/sandovc9/mambaforge/envs/pegasus/lib/libpython3.9.so
pythonhome:     /gstore/home/sandovc9/mambaforge/envs/pegasus:/gstore/home/sandovc9/mambaforge/envs/pegasus
version:        3.9.13 | packaged by conda-forge | (main, May 27 2022, 16:56:21)  [GCC 10.3.0]
numpy:          /gstore/home/sandovc9/mambaforge/envs/pegasus/lib/python3.9/site-packages/numpy
numpy_version:  1.23.3

NOTE: Python version was forced by use_python function
```

```
import pegasus as pg

ImportError: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /gstore/home/sandovc9/mambaforge/envs/pegasus/lib/python3.9/site-packages/scipy/linalg/_matfuncs_sqrtm_triu.cpython-39-x86_64-linux-gnu.so)
```

**Comment by yoshidk6** [comment]
I ran into a similar situation when I was trying to import xgboost in R with reticulate, while xgboost import was perfectly fine in Python REPL.
The solution that worked for me was downgrading xgboost from 1.6.x (which requires GLIBCXX_3.4.30) to 1.5.x. 

## TL;DR

This post (https://unix.stackexchange.com/a/557929) has nice explanations around this issue (what exactly are libstdc++.so.6, GLIBCXX_*, ...)

The problem seems to be stemming from different libstdc++.so.6 being called between R (& reticulate) and Conda/Python.
In your case `/usr/lib/x86_64-linux-gnu/libstdc++.so.6`, which I assume is the system default, is used by R & reticulate.
If you run `strings /usr/lib/x86_64-linux-gnu/libstdc++.so.6 | grep GLIBCXX`, you will see up to which version of GLIBCXX your libstdc++.so.6 supports. 
In my case it was up to GLIBCXX_3.4.28 and did not include GLIBCXX_3.4.30. This was the cause of the issue.

As far as I can see, there are couple of potential solutions. Only the last one worked for me.

1. Upgrade your system default libstd (https://github.com/lhelontra/tensorflow-on-arm/issues/13#issuecomment-489296444). I didn't have sudo access
2. Set LD_LIBRARY_PATH to tell R which exact library to use. This might not work according to this comment (https://github.com/rstudio/reticulate/issues/841#issuecomment-754967359)
3. Replace the system default libstdc++.so.6 with symbolic link to a newer version of libstdc++.so.6 somewhere in your system, like the one in conda (https://github.com/rstudio/reticulate/issues/841#issuecomment-1260791402). Again I didn't have sudo access
4.  Use an older version of Python packages that depends only on an older version of GLIBCXX


It would be super nice if reticulate package can detect this kind of incompatibility and flag this in a warning message.

**Comment by MKhGenesis** [comment]
https://stackoverflow.com/questions/73317676/importerror-usr-lib-aarch64-linux-gnu-libstdc-so-6-version-glibcxx-3-4-30
try this one

**Comment by ArnoZWang** [comment]
https://stackoverflow.com/questions/58424974/anaconda-importerror-usr-lib64-libstdc-so-6-version-glibcxx-3-4-21-not-fo
Have you ever tried this solution? It worked fine for me.

**Comment by HeyWeCome** [comment]
执行这个命令就可以了：

conda install -c conda-forge gcc=12.1.0

**Comment by t-kalinowski** [comment]
This is due to a binary incompatibility between conda and the R installation. 

The most straightforward solution is to switch from conda to virtual environments, which don't suffer from the same problem.

```r
library(reticulate)
install_python("3.10:latest")
virtualenv_create("r-scipy", version = "3.10", packages = "scipy")
```

(( The less straightforward solution is to have conda provide the R binary and all R packages. Note that this approach means that you can't use pre-built packages from CRAN or RStudio builds provided by Posit. ))

**Comment by mailmrcai** [comment]
**this command is ok:** conda install -c conda-forge gcc=12.1.0

**Comment by avishai987** [comment]
I also recommend to switch to virtual env, that's the only solution that solved the `'GLIBCXX_3.4.30' not found ` error 
(I also don't have sudo premission).
```
library(reticulate)
virtualenv_create("<your_path>/cnmf_dev", packages = <your_packages>,required = T)
use_virtualenv("<your_path>/cnmf_dev",required = T)
```

**Comment by LukeLIN-web** [comment]
> **this command is ok:** conda install -c conda-forge gcc=12.1.0此命令正常：conda install -c conda-forge gcc=12.1.0

Solving environment: unsuccessful initial attempt using frozen solve. Retrying with flexible solve.

**Comment by lumbric** [comment]
This issue is not specific to reticulate. I've written [a comprehensive overview on stackoverflow](https://stackoverflow.com/a/77940023/859591) explaining what is happening here and possible ways to fix it.

**Comment by WT215** [comment]
> I also recommend to switch to virtual env, that's the only solution that solved the `'GLIBCXX_3.4.30' not found ` error (I also don't have sudo premission).
> 
> ```
> library(reticulate)
> virtualenv_create("<your_path>/cnmf_dev", packages = <your_packages>,required = T)
> use_virtualenv("<your_path>/cnmf_dev",required = T)
> ```

I think this is the most robust way, particularly when you are working on a server.
